### ISSUE TITLE
Investigate why bfloat4_b returns different result from bfloat8_b/bfloat_16

### ISSUE THREAD

**Issue by bbradelTT** [tenstorrent/tt-metal#11954]
The following prints out a tensor with all values of 512 instead of 4096. 

```
import torch
import ttnn
import time
from tracy import Profiler
device_id = 0
device = ttnn.open_device(device_id=device_id)
a=ttnn.from_torch(1*torch.ones([1,1,512,4096], dtype=torch.bfloat16), dtype=ttnn.bfloat4_b, layout=ttnn.TILE_LAYOUT, device=device)
b=ttnn.from_torch(1*torch.ones([1,1,4096,6656], dtype=torch.bfloat16), dtype=ttnn.bfloat4_b, layout=ttnn.TILE_LAYOUT, device=device)
c = ttnn.matmul(a,b)
print(c)
ttnn.close_device(device)
exit()
```

Might be due to reduced accuracy of bfloat4_b. It would be good to understand the specific reason.

**Comment by bbradelTT** [comment]
bfloat4_b has
- 1 sign bit
- 3 mantissa bits
- Shared 8-bit exponent
Best accuracy would be up to 2^11 therefore would never get to 4k. 
This is due to reduced accuracy.

Closing
